src/ipc.rs

```diff
@@ -699,19 +691,16 @@ impl Debug for OpaqueIpcMessage {
     }
 }
 
-impl OpaqueIpcMessage {
-    fn new(
+impl IpcMessage {
+    pub(crate) fn new(
```

Comment:
I'm curious if there is reason for not marking this as `pub`. The struct only has pub fields, so it doesn't seem like we are gaining much in terms of encapsulation.

Reply:
Fair enough. I think mainly this wasn't marked as `pub` because it isn't expected that a user of `ipc-channel` would ever need to create this data structure. `pub(crate)` is useful in this case because the compiler can determine if the function ever becomes unused.

Reply:
> because it isn't expected that a user of ipc-channel would ever need to create this data structure

Should we then remove the 'pub' from the fields? Having them public means it will become part of our API in the next release and OpaqueIpcMessage only [had private fields](https://docs.rs/ipc-channel/latest/ipc_channel/ipc/struct.OpaqueIpcMessage.html). I'm happy to do the change.

Reply:
I am not opposed to making the fields private again.